### PR TITLE
Implemented Frontend Unit Test Cases for Upload file Functionality ( #47) Fixed

### LINKED ISSUE #47: Add the frontend unit test cases for Upload file functionality
With the change of the frontend technology from Angular to React, all the existing frontend test cases became irrelevant.
Write the unit test cases for the file upload functionality using the jest framework.

Cover the following scenarios:
1. Clicking on Choose File and uploading an ORCA file
2. Clicking on Choose File and try to upload any file other than the ORCA file
3. Clicking on Preview without uploading any file and validating the error message
4. Clicking on Download without uploading any file and validating the error message

Acceptance criteria:
All the test cases should pass and the tests should match the functionality.

### PR DESCRIPTION
**Pull request by mkalal6**
Fixes #47

**What was changed?**

*I added the key file uploadFile.test.js for unit testing the file upload functionality. The uploadFile.test.js includes unit tests for uploading an ORCA file, ensuring only valid file types are uploaded, and preventing non-ORCA file uploads.*

**Why was it changed?**

*The transition from Angular to React rendered the old frontend test cases irrelevant. This change is part of the effort to create unit tests for the new React components using Jest. Specifically, the file upload functionality needed to be tested, ensuring compliance with the requirements for ORCA file uploads. By adding unit tests, we ensure that the feature behaves as expected in various scenarios, such as uploading valid/invalid files or triggering an error when no file is uploaded.*

**How was it changed?**

*The uploadFile.test.js file was created to test ORCA file uploads, including both valid and invalid file scenarios, using Jest and React Testing Library. For now, the 'Clicking on Preview without uploading any file and validating the error message' test is commented out as there is no preview button yet. The "npm test" can be run on the terminal to run the tests.*

### CODE REVIEW

File: client-app/src/_test_/uploadFile.test.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    fireEvent.click(uploadButton);
+
+    // Assert that an error message is displayed
+    await waitFor(() => {
+      expect(screen.getByText('Only ORCA files are allowed')).toBeInTheDocument();
+    });
+  });
+
+
+});
```
Comment (Ebruns4) [inline client-app/src/_test_/uploadFile.test.js:55]: Clean and efficient push

**Review by Lyskawa-Aonyx — COMMENTED** [review]
Great work! Everything looks good and I can't find any bugs with it!

File: client-app/package-lock.json [hunk 2]
```diff
@@ -20,6 +20,12 @@
         "react-router-dom": "^6.23.0",
         "react-scripts": "5.0.1",
         "web-vitals": "^2.1.4"
+      },
```
[the remark this answers is not in the document]
Comment (mkalal6) [reply client-app/package-lock.json:23]: Yes I changed the json file as well.

**Comment by kate-holdener** [comment]
How do you run these tests?

**Comment by SiriChandanaGarimella** [comment]
@mkalal6 - Great work! As we talked about, I have committed the code with a working unit test case for the first scenario. Please pull the latest code, test it once, and add the other test cases. 
I have removed jest.config.js and babel.config.js files and updated the package.json file. So, please update the description of the PR accordingly. 

Also, please update the README.md file with the command to run the frontend test cases - npm test

**Comment by SiriChandanaGarimella** [comment]
Great work! The tests are running successfully. I'll be adding a GitHub Actions workflow to automate these tests with every push (as part of issue #80)

**Review by SiriChandanaGarimella — APPROVED** [review]
lgtm;

**Pull request merged**